modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -1816,6 +1821,27 @@ public CheckpointUrgency checkpointUrgency() {
         return checkpointUrgency.get();
     }
 
+    /**
+     * Checks if region is sufficiently full.
+     *
+     * @param dirtyRatioThreshold Max allowed dirty pages ration.
+     */
+    public boolean shouldThrottle(double dirtyRatioThreshold) {
+        Segment[] segments = this.segments;
+
+        if (segments == null) {
+            return false;
+        }
+
+        for (Segment segment : segments) {
```

Comment:
I think that iteration over the counting loop will be performed faster and without additional iterator allocation.

Reply:
It's an array, there's no allocation of iterators here